src/tests/multihost/alltests/test_inmemory_cache.py

```diff
@@ -0,0 +1,329 @@
+            4. if the sss_nss process is not running, invalidate the memcache.
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+            4. Should succeed
+        """
+        client = multihost.client[0]
+        log_sssd = f'/var/log/sssd/sssd_{ds_instance_name}.log'
+        client.run_command("service sssd stop")
+        client.run_command("sss_cache -U --domain=example1")
+        client.run_command("sss_cache -G --domain=example1")
+        client.run_command("service sssd start")
+        clear_only_domain_log(multihost)
+        client.run_command("getent passwd foo1")
+        client.run_command("getent group ldapusers")
+        find_logs(multihost, log_sssd, "Got request for ")
+        find_logs(multihost, log_sssd, "uid=foo1")
+        find_logs(multihost, log_sssd, "cn=ldapusers")
+
+    @staticmethod
+    def test_memcache(multihost):
+        """
+        :title: Set value of option memcache timeout to 0 and expire InMemory cache immediately
+        :id: 19dfa7ae-96e9-11ed-adf3-845cf3eff344
+        :steps:
+            1. Set value of option memcache timeout to 0 and expire InMemory cache immediately
+            2. Validate the user using sss cache
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+        """
+        client = multihost.client[0]
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'memcache_timeout': '0'}, action='update')
+        tools.clear_sssd_cache()
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(2)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(5)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(5)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+
+    @staticmethod
+    def test_bz1484376(multihost):
+        """
+        :title: Add a configuration option to SSSD to disable the memory cache bz1484376
+                Set value of option Memcache timeout to 35
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1484376
+        :id: b64237b4-96ef-11ed-adb1-845cf3eff344
+        :steps:
+            1. Add a configuration option to SSSD to disable the memory cache
+            2. Set memcache_timeout=0
+            3. Set value of option Memcache timeout to 35
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        client = multihost.client[0]
+        client.run_command("rm -rvf /var/lib/sss/mc/*")
```

Comment:
sssdTools(multihost.client[0]).clear_sssd_cache()

Reply:
sssdTools(multihost.client[0]).clear_sssd_cache()    will clear logs as well , here we need to remove only /var/lib/sss/mc/* not logs as well

Reply:
Here we need to test /var/lib/sss/mc/* was empty after sssd restart . clear_sssd_cache will clear other stuff as well which we do not need in this test